Core/OfficeDevPnP.Core/Sites/SiteCollection.cs

```diff
@@ -654,6 +654,45 @@ public static async Task<string> TeamifySiteAsync(ClientContext context)
                 return await Task.Run(() => responseString);
             }
         }
+
+        /// <summary>
+        /// Deletes a Communication site or a Group-less Modern team site.
+        /// </summary>
+        /// <param name="context">Context to operate against</param>
+        /// <returns></returns>
+        public static async Task<bool> DeleteSiteAsync(ClientContext context)
+        {
+            bool siteDeleted = false;
+
+            var webTemplateId = context.Web.GetBaseTemplateId();
+
+            context.Site.EnsureProperties(s => s.Id, s => s.GroupId, s => s.Url);
+
+            if (webTemplateId == "SITEPAGEPUBLISHING#0" || webTemplateId == "STS#3")
+            {
+                var result = await context.Web.ExecutePost("/_api/SPSiteManager/delete", $@" {{ ""siteId"": ""{context.Site.Id.ToString()}"" }}");
+
+                var parsedResult = JObject.Parse(result);
+
+                siteDeleted = Convert.ToBoolean(parsedResult["odata.null"]);
+
+                return await Task.Run(() => siteDeleted);
+            }
+            else if (webTemplateId == "GROUP#0" || context.Site.GroupId != Guid.Empty)
+            {
+                var result = await context.Web.ExecutePost($"/_api/GroupSiteManager/Delete?siteUrl='{context.Site.Url}'", string.Empty);
+
+                var parsedResult = JObject.Parse(result);
+
+                siteDeleted = Convert.ToBoolean(parsedResult["odata.null"]);
+
+                return await Task.Run(() => siteDeleted);
+            }
+            else
+            {
+                throw new Exception("Only deletion of Communication site or Modern team site is supported by this method.");
```

Comment:
Why not automatically delete "old" site?
Maybe make a DeleteModernSite that only deletes a Modern site and a general DeleteSite that always deletes the site regardless of the type of site (and linked group).

Reply:
You can already delete the sites of all templates using TenantExtensions. This PR adds an additional method in the Site collection context instead of Tenant context. Also, we are using the newer APIs which were specifically made for Modern sites.